### ISSUE TITLE
Suggestion: Providing a way to generate connectable boxes

### ISSUE THREAD

**Issue by DannyBen** [piotrmurach/tty-box#3]
I see there is some preparation in the code for borders other than the four corners:

https://github.com/piotrmurach/tty-box/blob/94ebd8653e23d7e1b6f20d424ec419f7021de401/lib/tty/box.rb#L14-L17

But I could not find anything in the code to let me use these cross connectors and dividers.

Is there currently a way to draw boxes that connect nicely to each other, like this:

```
╔════════╦════════╗
║left    ║right   ║
╚════════╩════════╝
```

Instead of like this:

```
╔════════╔════════╗
║left    ║right   ║
╚════════╚════════╝
```

If there is none, I would like to cast my "+1" for having it.

I am sure you have already thought about it, but just in case, I would like to suggest one of two interfaces for the user to specify these borders:

### Option 1: More intuitive, less flexible, probably more work

By specifying only the side of the box that should be connected, like:

```ruby
puts TTY::Box.frame(top: 0, left: 9, width: 10, height: 3. connect: :left) { "right" }
```

will produce:
```
┬────────┐
│right   │
┴────────┘
```

and `connect: [:left, :top]` will produce:

```
┼────────┤
│right   │
┴────────┘
```

### Option 2: Less intuitive, more flexible, probably less work

Instead of using the "magic" `connect`, maybe let the user specify explicitly which corners they need:

```ruby
options = {
  top: 0, left: 9, width: 10, height: 3,
  corner1: :cross,
  corner4: :left_divider
}

# corners are indexed clockwise, starting from top left
# values can use the same names as the `*_char` methods

puts TTY::Box.frame(options) { "right" }
```

**Comment by piotrmurach** [comment]
Hi Danny,

Thank you for this issue!

This is an extraction from a terminal game I've coded. It matched my requirements so I've cut an initial release. However, I do like tty components to be generalised and thought this will need to be flexible so as to allow construction of complex box layouts, similar to commander application - my childhood favourite 😄 

Therefore, I was thinking about a single `:border` attribute that currently only allows to switch on and off main 4 borders (see [border parser](https://github.com/piotrmurach/tty-box/blob/master/lib/tty/box/border.rb)). I'm keen on great user experience and wouldn't like people having to count corners in their head. Similarly, the connect doesn't seem to address middle of the border dividers etc... Having said that, I'm sure tty-box could be extended to include your current thinking. For example, using the current `:border` attribute we could add more areas of the border with defaults: 

```ruby
options = {
  top: 0, left: 9, width: 10, height: 3,
  border: {
    top: :line, # doesn't need to be specified as default :line
    top_left: :cross, # or :corner
    top_mid: :divider, # or :line
    top_right: :cross # or :corner or the :triple thingy?
    etc...
  }
}
``` 

What do you think? Do you have time to submit PR we could be working towards?

**Comment by DannyBen** [comment]
Yes: The border attribute seems like the right place for this.
Maybe: I will have time to attempt implementation.

**Comment by piotrmurach** [comment]
Implemented and released `v0.3.0`. Please see [2.5 border docs](https://github.com/piotrmurach/tty-box#25-border)
